api/src/database/migrations/20240924A-migrate-legacy-comments.ts

```diff
@@ -0,0 +1,60 @@
+import { Action } from '@directus/constants';
+import type { Knex } from 'knex';
+import { randomUUID } from 'node:crypto';
+
+export async function up(knex: Knex): Promise<void> {
+	const rowsLimit = 50;
+	let hasMore = true;
+
+	while (hasMore) {
+		const legacyComments = await knex
+			.select('*')
+			.from('directus_activity')
+			.where('action', '=', Action.COMMENT)
+			.limit(rowsLimit);
+
+		if (legacyComments.length === 0) {
+			hasMore = false;
+			break;
+		}
+
+		await knex.transaction(async (trx) => {
+			for (const legacyComment of legacyComments) {
+				let primaryKey;
+
+				// Migrate legacy comment
+				if (legacyComment['action'] === Action.COMMENT) {
+					primaryKey = randomUUID();
+
+					await trx('directus_comments').insert({
+						id: primaryKey,
+						collection: legacyComment.collection,
+						item: legacyComment.item,
+						comment: legacyComment.comment,
+						user_created: legacyComment.user,
+						date_created: legacyComment.timestamp,
+					});
+
+					await trx('directus_activity')
+						.update({
+							action: Action.CREATE,
+							collection: 'directus_comments',
+							item: primaryKey,
+							comment: null,
+						})
+						.where('id', '=', legacyComment.id);
+				}
+			}
+		});
+	}
+
+	await knex.schema.alterTable('directus_activity', (table) => {
+		table.dropColumn('comment');
+	});
+}
+
+export async function down(knex: Knex): Promise<void> {
+	await knex.schema.alterTable('directus_activity', (table) => {
+		table.text('comment');
+	});
```

Comment:
Technically this doesnt really revert the migration because the comments will be missing inside the activity table, but I think we aligned on us not providing that anyway -- right? @ComfortablyCoding

Reply:
The previous migration should cover the down aspect. 🤔 
https://github.com/directus/directus/blob/adf5171336dfc821aed1cfc56d872867d56f6a55/api/src/database/migrations/20240909A-separate-comments.ts#L26-L76

Reply:
That is correct @DanielBiegler, as per @licitdev this is handled in the previous migration so no need to do here.